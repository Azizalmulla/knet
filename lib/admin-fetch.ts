export async function adminFetch(path: string, init: RequestInit = {}) {
  const token = typeof window !== 'undefined' ? sessionStorage.getItem('admin_token') : null;
  
  const res = await fetch(path, { 
    ...init, 
    headers: { 
      ...(init.headers || {}), 
      Authorization: `Bearer ${token || ''}` 
    } 
  });
  
  if (!res.ok) {
    // Handle 401 unauthorized - session expired
    if (res.status === 401 && typeof window !== 'undefined') {
      sessionStorage.removeItem('admin_token');
      
      // Show toast notification if available
      if (window.location.pathname.startsWith('/admin')) {
        // Create a simple toast notification
        const toast = document.createElement('div');
        toast.textContent = 'Session expiredâ€”please log in again.';
        toast.style.cssText = `
          position: fixed; top: 20px; right: 20px; z-index: 9999;
          background: #f56565; color: white; padding: 12px 24px;
          border-radius: 8px; font-size: 14px; font-weight: 500;
          box-shadow: 0 4px 12px rgba(0,0,0,0.15);
        `;
        document.body.appendChild(toast);
        
        // Auto-remove toast and reload page after 3 seconds
        setTimeout(() => {
          toast.remove();
          window.location.reload();
        }, 3000);
      }
    }
    
    throw new Error(`ADMIN_FETCH_${res.status}`);
  }
  
  return res.json();
}
