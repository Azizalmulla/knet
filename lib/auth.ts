import NextAuth from "next-auth"
import { authConfig } from "./auth.config"

export const { handlers, auth, signIn, signOut } = NextAuth(authConfig)

// Export a wrapper for getServerSession for compatibility
export const getServerSession = auth
