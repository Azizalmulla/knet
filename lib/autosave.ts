/**
 * Local draft autosave utility with debouncing for AI CV Builder
 */

import { useEffect, useRef, useCallback } from 'react';
import { UseFormWatch } from 'react-hook-form';

const AUTOSAVE_KEY = 'knet_cv_draft';
const AUTOSAVE_DEBOUNCE_MS = 5000; // 5 seconds
const AUTOSAVE_DISABLED = process.env.NEXT_PUBLIC_DISABLE_AUTOSAVE === 'true';

export interface AutosaveOptions {
  disabled?: boolean;
  debounceMs?: number;
  storageKey?: string;
}

/**
 * Hook for autosaving form data to localStorage with debouncing
 */
export function useAutosave(
  watch: UseFormWatch<any>,
  options: AutosaveOptions = {}
) {
  const {
    disabled = AUTOSAVE_DISABLED,
    debounceMs = AUTOSAVE_DEBOUNCE_MS,
    storageKey = AUTOSAVE_KEY
  } = options;

  const timeoutRef = useRef<NodeJS.Timeout>();
  const lastSavedRef = useRef<string>('');

  const saveToStorage = useCallback((data: any) => {
    if (disabled || typeof window === 'undefined') return;

    try {
      const serialized = JSON.stringify({
        data,
        timestamp: Date.now(),
        version: '1.0'
      });

      // Only save if data has actually changed
      if (serialized !== lastSavedRef.current) {
        localStorage.setItem(storageKey, serialized);
        lastSavedRef.current = serialized;
        console.log('üîÑ Draft autosaved at', new Date().toLocaleTimeString());
      }
    } catch (error) {
      console.warn('Failed to save draft:', error);
    }
  }, [disabled, storageKey]);

  const debouncedSave = useCallback((data: any) => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current);
    }

    timeoutRef.current = setTimeout(() => {
      saveToStorage(data);
    }, debounceMs);
  }, [saveToStorage, debounceMs]);

  useEffect(() => {
    if (disabled) return;

    const subscription = watch((data) => {
      // Skip if data is empty or only has default values
      if (!data || Object.keys(data).length === 0) return;
      
      // Skip if no meaningful data (just empty strings)
      const hasContent = Object.values(data).some(value => {
        if (typeof value === 'string') return value.trim().length > 0;
        if (Array.isArray(value)) return value.length > 0;
        if (value && typeof value === 'object') return Object.keys(value).length > 0;
        return Boolean(value);
      });

      if (hasContent) {
        debouncedSave(data);
      }
    });

    return () => {
      subscription.unsubscribe();
      if (timeoutRef.current) {
        clearTimeout(timeoutRef.current);
      }
    };
  }, [watch, debouncedSave, disabled]);

  // Cleanup on unmount
  useEffect(() => {
    return () => {
      if (timeoutRef.current) {
        clearTimeout(timeoutRef.current);
      }
    };
  }, []);
}

/**
 * Load saved draft from localStorage
 */
export function loadDraft(storageKey: string = AUTOSAVE_KEY): any | null {
  // Check environment variable dynamically for tests
  const isDisabled = process.env.NEXT_PUBLIC_DISABLE_AUTOSAVE === 'true';
  if (isDisabled || typeof window === 'undefined') return null;

  try {
    const saved = localStorage.getItem(storageKey);
    if (!saved) return null;

    const parsed = JSON.parse(saved);
    
    // Check if draft is not too old (24 hours)
    const maxAge = 24 * 60 * 60 * 1000; // 24 hours
    if (Date.now() - parsed.timestamp > maxAge) {
      localStorage.removeItem(storageKey);
      return null;
    }

    return parsed.data;
  } catch (error) {
    console.warn('Failed to load draft:', error);
    return null;
  }
}

/**
 * Clear saved draft from localStorage
 */
export function clearDraft(storageKey: string = AUTOSAVE_KEY): void {
  if (typeof window === 'undefined') return;

  try {
    localStorage.removeItem(storageKey);
    console.log('üóëÔ∏è  Draft cleared');
  } catch (error) {
    console.warn('Failed to clear draft:', error);
  }
}

/**
 * Check if there's a saved draft available
 */
export function hasDraft(storageKey: string = AUTOSAVE_KEY): boolean {
  return loadDraft(storageKey) !== null;
}

/**
 * Get draft metadata (timestamp, etc.)
 */
export function getDraftInfo(storageKey: string = AUTOSAVE_KEY): { timestamp: number; age: string } | null {
  if (AUTOSAVE_DISABLED || typeof window === 'undefined') return null;

  try {
    const saved = localStorage.getItem(storageKey);
    if (!saved) return null;

    const parsed = JSON.parse(saved);
    const timestamp = parsed.timestamp;
    const ageMs = Date.now() - timestamp;
    const ageMinutes = Math.floor(ageMs / (1000 * 60));
    
    let age: string;
    if (ageMinutes < 1) {
      age = 'less than a minute ago';
    } else if (ageMinutes < 60) {
      age = `${ageMinutes} minute${ageMinutes === 1 ? '' : 's'} ago`;
    } else {
      const ageHours = Math.floor(ageMinutes / 60);
      age = `${ageHours} hour${ageHours === 1 ? '' : 's'} ago`;
    }

    return { timestamp, age };
  } catch (error) {
    return null;
  }
}
