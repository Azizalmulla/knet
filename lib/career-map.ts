// Exact mapping provided by the USER. Do NOT alter wording.
export const careerMapRows = [
  {
    "Field of Study": "Business Management",
    "Area of Interest": "Operations",
    "Suggested Vacancies": "Bank Operations"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Operations",
    "Suggested Vacancies": "Payment Operations/Core Operations"
  },
  {
    "Field of Study": "Business Management",
    "Area of Interest": "Customer Care",
    "Suggested Vacancies": "Customer Care/Disputes Management/Customer Experience Management"
  },
  {
    "Field of Study": "Business Management",
    "Area of Interest": "Business Development",
    "Suggested Vacancies": "Business Relationship"
  },
  {
    "Field of Study": "Media/Marketing/PR",
    "Area of Interest": "Marketing",
    "Suggested Vacancies": "Marketing"
  },
  {
    "Field of Study": "Finance and Accounting",
    "Area of Interest": "Digital Transformation & Innovation",
    "Suggested Vacancies": "Business Intellegence"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Digital Transformation & Innovation",
    "Suggested Vacancies": "Innovation/Digital Transformation"
  },
  {
    "Field of Study": "Business Management",
    "Area of Interest": "Digital Transformation & Innovation",
    "Suggested Vacancies": "Innovation/Digital Transformation/Business Intellegence"
  },
  {
    "Field of Study": "Finance and Accounting",
    "Area of Interest": "Finance\u00a0",
    "Suggested Vacancies": "Finance/Accounting"
  },
  {
    "Field of Study": "Finance and Accounting",
    "Area of Interest": "Supply Chain",
    "Suggested Vacancies": "Procurement and Inventory Management"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Supply Chain",
    "Suggested Vacancies": "Vendor Management"
  },
  {
    "Field of Study": "Business Management",
    "Area of Interest": "HR",
    "Suggested Vacancies": "Human Resources"
  },
  {
    "Field of Study": "Finance and Accounting",
    "Area of Interest": "HR",
    "Suggested Vacancies": "Human Resources"
  },
  {
    "Field of Study": "Business Management",
    "Area of Interest": "Project Management\u00a0",
    "Suggested Vacancies": "Project Development"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Project Management\u00a0",
    "Suggested Vacancies": "Project Development"
  },
  {
    "Field of Study": "Finance and Accounting",
    "Area of Interest": "Strategy",
    "Suggested Vacancies": "Strategy"
  },
  {
    "Field of Study": "Business Management",
    "Area of Interest": "Strategy",
    "Suggested Vacancies": "Strategy"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Audit",
    "Suggested Vacancies": "IT Aduit"
  },
  {
    "Field of Study": "Finance and Accounting",
    "Area of Interest": "Audit",
    "Suggested Vacancies": "Finance Audit"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Risk Management",
    "Suggested Vacancies": "Operational Risk/technology Risk/Business Continuity"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Information Security",
    "Suggested Vacancies": "Information Security/Cyber Security"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Fraud Management ",
    "Suggested Vacancies": "Transaction Fraud/Enterprise Fraud"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "IT",
    "Suggested Vacancies": "Development/System Excellence/Application Support/Network Management/Data & Server Management/IT Support Center"
  },
  {
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Area of Interest": "Security Operations ",
    "Suggested Vacancies": "Network Security/Application Security"
  },
  {
    "Field of Study": "Others ",
    "Area of Interest": "(as per the ebove)",
    "Suggested Vacancies": "(to be as er the area of interest and suggested vacancy)"
  }
];

export function getFields(): string[] {
  return Array.from(new Set(careerMapRows.map(r => r["Field of Study"])));
}

export function getAreasForField(field: string): string[] {
  return Array.from(
    new Set(
      careerMapRows
        .filter(r => r["Field of Study"] === field)
        .map(r => r["Area of Interest"]) 
    )
  );
}

export function matchSuggestedVacancies(field: string, area: string): string | null {
  const row = careerMapRows.find(r => r["Field of Study"] === field && r["Area of Interest"] === area);
  return row ? row["Suggested Vacancies"] : null;
}

export function findRowForAudit(field: string, area: string): typeof careerMapRows[0] | null {
  return careerMapRows.find(r => r["Field of Study"] === field && r["Area of Interest"] === area) || null;
}
