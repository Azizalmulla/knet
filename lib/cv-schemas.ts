import { z } from 'zod';

// Personal Info Step
export const personalInfoSchema = z.object({
  fullName: z.string().min(1, 'Full name is required'),
  email: z.string().email('Invalid email address'),
  phone: z.string().min(1, 'Phone number is required'),
  location: z.string().min(1, 'Location is required'),
  summary: z.string().optional(),
});

// Education Step
export const educationSchema = z.object({
  education: z.array(z.object({
    institution: z.string().min(1, 'Institution is required'),
    degree: z.string().min(1, 'Degree is required'),
    field: z.string().min(1, 'Field of study is required'),
    startDate: z.string().min(1, 'Start date is required'),
    endDate: z.string().optional(),
    gpa: z.string().optional(),
    description: z.string().optional(),
  })).min(1, 'At least one education entry is required'),
});

// Experience Step
export const experienceSchema = z.object({
  experience: z.array(z.object({
    company: z.string().min(1, 'Company is required'),
    position: z.string().min(1, 'Position is required'),
    startDate: z.string().min(1, 'Start date is required'),
    endDate: z.string().optional(),
    current: z.boolean().default(false),
    description: z.string().optional(),
    bullets: z.array(z.string()).default([]),
  })).default([]),
});

// Projects Step
export const projectsSchema = z.object({
  projects: z.array(z.object({
    name: z.string().min(1, 'Project name is required'),
    description: z.string().min(1, 'Description is required'),
    technologies: z.array(z.string()).default([]),
    url: z.string().optional(),
    bullets: z.array(z.string()).default([]),
  })).default([]),
});

// Skills Step
export const skillsSchema = z.object({
  skills: z.object({
    technical: z.array(z.string()).default([]),
    languages: z.array(z.string()).default([]),
    soft: z.array(z.string()).default([]),
  }),
});

// Complete CV Schema
export const cvSchema = z.object({
  ...personalInfoSchema.shape,
  ...educationSchema.shape,
  ...experienceSchema.shape,
  ...projectsSchema.shape,
  ...skillsSchema.shape,
  template: z.enum(['minimal', 'modern', 'creative']).default('minimal'),
  language: z.enum(['en', 'ar']).default('en'),
});

export type PersonalInfo = z.infer<typeof personalInfoSchema>;
export type Education = z.infer<typeof educationSchema>;
export type Experience = z.infer<typeof experienceSchema>;
export type Projects = z.infer<typeof projectsSchema>;
export type Skills = z.infer<typeof skillsSchema>;
export type CVData = z.infer<typeof cvSchema>;
