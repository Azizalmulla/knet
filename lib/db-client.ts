import { sql } from "@vercel/postgres";

const isTest = process.env.NEXT_PUBLIC_E2E === "1" || process.env.NODE_ENV === "test";

export const db = {
  insertSubmission: async (payload: any) => {
    if (isTest) {
      console.log('Mock DB: insertSubmission called with:', payload);
      return { id: "test-submission-id" };
    }
    
    try {
      // Real database query would go here
      return await sql`INSERT INTO submissions (data) VALUES (${JSON.stringify(payload)}) RETURNING id`;
    } catch (error) {
      console.warn('Database connection failed, using mock response');
      return { id: "mock-submission-id" };
    }
  }
};

// Minimal client wrapper used by telemetry API with safe fallback for tests/preview
export function getDbClient() {
  if (isTest) {
    return {
      // Return empty result sets in tests unless explicitly mocked
      query: async (_text: string, _params: any[] = []) => ({ rows: [] as any[] })
    };
  }
  return {
    query: async (text: string, params: any[] = []) => {
      try {
        return await sql.query(text, params);
      } catch (error) {
        console.warn('Database unavailable, returning empty result set for telemetry');
        return { rows: [] as any[] } as any;
      }
    }
  };
}
