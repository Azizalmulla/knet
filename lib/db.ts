import { sql } from '@vercel/postgres';

export const db = sql;

// The following is the SQL to create the table in Vercel Postgres dashboard:

// CREATE TABLE students (
//   id SERIAL PRIMARY KEY,
//   full_name VARCHAR(255) NOT NULL,
//   email VARCHAR(255) NOT NULL UNIQUE,
//   phone VARCHAR(255),
//   field_of_study VARCHAR(255) NOT NULL,
//   area_of_interest VARCHAR(255) NOT NULL,
//   cv_type VARCHAR(50) NOT NULL, // 'uploaded' or 'ai'
//   cv_url VARCHAR(500),
//   submitted_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
// );

// For blob, we'll use @vercel/blob

// Also, perhaps a table for vacancies, but since it's from JSON, maybe not needed.
