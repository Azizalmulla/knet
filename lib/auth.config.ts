import { NextAuthConfig } from "next-auth"
import GoogleProvider from "next-auth/providers/google"
import MicrosoftEntraId from "next-auth/providers/microsoft-entra-id"
import EmailProvider from "next-auth/providers/email"
import { sql } from '@/lib/db'

// Helper to send email via Resend
async function sendVerificationRequest({
  identifier: email,
  url,
  provider,
}: {
  identifier: string
  url: string
  provider: any
}) {
  const { host } = new URL(url)
  const res = await fetch("https://api.resend.com/emails", {
    method: "POST",
    headers: {
      Authorization: `Bearer ${process.env.RESEND_API_KEY}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      from: process.env.RESEND_FROM || "Careerly <onboarding@resend.dev>",
      to: [email],
      subject: `Sign in to Careerly`,
      html: `
        <!DOCTYPE html>
        <html>
          <head>
            <meta name="viewport" content="width=device-width, initial-scale=1.0">
            <style>
              body { font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif; }
              .container { max-width: 600px; margin: 0 auto; padding: 20px; background: #0a0a0a; color: #ffffff; }
              .header { text-align: center; padding: 30px 0; }
              .logo { font-size: 32px; font-weight: bold; color: #ffffff; }
              .content { background: #1a1a1a; border-radius: 12px; padding: 30px; margin: 20px 0; }
              .button { display: inline-block; padding: 12px 24px; background: #ffffff; color: #000000; text-decoration: none; border-radius: 8px; font-weight: 600; margin: 20px 0; }
              .footer { text-align: center; color: #666; font-size: 12px; padding: 20px 0; }
              a { color: #ffffff; }
            </style>
          </head>
          <body>
            <div class="container">
              <div class="header">
                <div class="logo">Careerly</div>
              </div>
              <div class="content">
                <h2 style="margin-top: 0;">Sign in to your account</h2>
                <p>Click the button below to sign in to Careerly. This link will expire in 15 minutes.</p>
                <div style="text-align: center;">
                  <a href="${url}" class="button">Sign in to Careerly</a>
                </div>
                <p style="color: #888; font-size: 14px;">If you didn't request this email, you can safely ignore it.</p>
              </div>
              <div class="footer">
                <p>Â© ${new Date().getFullYear()} Careerly. All rights reserved.</p>
              </div>
            </div>
          </body>
        </html>
      `,
      text: `Sign in to Careerly\n\nClick the link below to sign in:\n${url}\n\nThis link will expire in 15 minutes.\n\nIf you didn't request this email, you can safely ignore it.`,
    }),
  })

  if (!res.ok) {
    throw new Error(`Email send failed: ${res.statusText}`)
  }
}

export const authConfig: NextAuthConfig = {
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID!,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
    }),
    MicrosoftEntraId({
      clientId: process.env.AZURE_AD_CLIENT_ID!,
      clientSecret: process.env.AZURE_AD_CLIENT_SECRET!,
    }),
    // EmailProvider temporarily disabled - requires database adapter
    // Will re-enable with proper adapter configuration
  ],
  pages: {
    signIn: "/student/login",
    error: "/student/login",
    verifyRequest: "/student/verify",
  },
  session: {
    strategy: "jwt",
    maxAge: 30 * 24 * 60 * 60, // 30 days
  },
  callbacks: {
    async signIn({ user, account, profile }) {
      // Upsert student user
      if (user.email) {
        try {
          await sql`
            INSERT INTO student_users (email, email_lc, name, avatar_url)
            VALUES (${user.email}, ${user.email.toLowerCase()}, ${user.name}, ${user.image})
            ON CONFLICT (email) DO UPDATE
            SET name = COALESCE(EXCLUDED.name, student_users.name),
                avatar_url = COALESCE(EXCLUDED.avatar_url, student_users.avatar_url)
          `
        } catch (error) {
          console.error("Failed to upsert student user:", error)
        }
      }
      return true
    },
    async jwt({ token, user, account }) {
      if (user) {
        token.uid = user.id
        token.email = user.email
        token.name = user.name
        token.picture = user.image
        token.roles = ["student"]
      }
      return token
    },
    async session({ session, token }) {
      if (token && session.user) {
        session.user.id = token.uid as string
        session.user.email = token.email as string
        session.user.name = token.name as string
        session.user.image = token.picture as string
        ;(session as any).roles = token.roles || ["student"]
      }
      return session
    },
  },
}
