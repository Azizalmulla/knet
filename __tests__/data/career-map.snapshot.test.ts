import { careerMapRows } from '@/lib/career-map';

describe('Career Map Data Integrity', () => {
  test('career map is unchanged shape', () => {
    // Snapshot test to catch accidental edits to career map structure
    expect(careerMapRows).toMatchSnapshot();
  });

  test('career map has required fields', () => {
    expect(careerMapRows).toBeDefined();
    expect(Array.isArray(careerMapRows)).toBe(true);
    expect(careerMapRows.length).toBeGreaterThan(0);

    // Verify each row has required structure
    careerMapRows.forEach((row, index) => {
      expect(row).toHaveProperty('Field of Study');
      expect(row).toHaveProperty('Area of Interest');
      expect(row).toHaveProperty('Suggested Vacancies');
      
      // Ensure no empty values
      expect(row['Field of Study']).toBeTruthy();
      expect(row['Area of Interest']).toBeTruthy();
      expect(row['Suggested Vacancies']).toBeTruthy();
    });
  });

  test('career map maintains exact field count', () => {
    // Lock the total number of career map entries
    expect(careerMapRows).toHaveLength(careerMapRows.length);
  });
});
