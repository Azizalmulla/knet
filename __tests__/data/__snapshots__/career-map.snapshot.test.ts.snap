// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Career Map Data Integrity career map is unchanged shape 1`] = `
[
  {
    "Area of Interest": "Operations",
    "Field of Study": "Business Management",
    "Suggested Vacancies": "Bank Operations",
  },
  {
    "Area of Interest": "Operations",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Payment Operations/Core Operations",
  },
  {
    "Area of Interest": "Customer Care",
    "Field of Study": "Business Management",
    "Suggested Vacancies": "Customer Care/Disputes Management/Customer Experience Management",
  },
  {
    "Area of Interest": "Business Development",
    "Field of Study": "Business Management",
    "Suggested Vacancies": "Business Relationship",
  },
  {
    "Area of Interest": "Marketing",
    "Field of Study": "Media/Marketing/PR",
    "Suggested Vacancies": "Marketing",
  },
  {
    "Area of Interest": "Digital Transformation & Innovation",
    "Field of Study": "Finance and Accounting",
    "Suggested Vacancies": "Business Intellegence",
  },
  {
    "Area of Interest": "Digital Transformation & Innovation",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Innovation/Digital Transformation",
  },
  {
    "Area of Interest": "Digital Transformation & Innovation",
    "Field of Study": "Business Management",
    "Suggested Vacancies": "Innovation/Digital Transformation/Business Intellegence",
  },
  {
    "Area of Interest": "Finance ",
    "Field of Study": "Finance and Accounting",
    "Suggested Vacancies": "Finance/Accounting",
  },
  {
    "Area of Interest": "Supply Chain",
    "Field of Study": "Finance and Accounting",
    "Suggested Vacancies": "Procurement and Inventory Management",
  },
  {
    "Area of Interest": "Supply Chain",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Vendor Management",
  },
  {
    "Area of Interest": "HR",
    "Field of Study": "Business Management",
    "Suggested Vacancies": "Human Resources",
  },
  {
    "Area of Interest": "HR",
    "Field of Study": "Finance and Accounting",
    "Suggested Vacancies": "Human Resources",
  },
  {
    "Area of Interest": "Project Management ",
    "Field of Study": "Business Management",
    "Suggested Vacancies": "Project Development",
  },
  {
    "Area of Interest": "Project Management ",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Project Development",
  },
  {
    "Area of Interest": "Strategy",
    "Field of Study": "Finance and Accounting",
    "Suggested Vacancies": "Strategy",
  },
  {
    "Area of Interest": "Strategy",
    "Field of Study": "Business Management",
    "Suggested Vacancies": "Strategy",
  },
  {
    "Area of Interest": "Audit",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "IT Aduit",
  },
  {
    "Area of Interest": "Audit",
    "Field of Study": "Finance and Accounting",
    "Suggested Vacancies": "Finance Audit",
  },
  {
    "Area of Interest": "Risk Management",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Operational Risk/technology Risk/Business Continuity",
  },
  {
    "Area of Interest": "Information Security",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Information Security/Cyber Security",
  },
  {
    "Area of Interest": "Fraud Management ",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Transaction Fraud/Enterprise Fraud",
  },
  {
    "Area of Interest": "IT",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Development/System Excellence/Application Support/Network Management/Data & Server Management/IT Support Center",
  },
  {
    "Area of Interest": "Security Operations ",
    "Field of Study": "Computer Engineering/Computer Science/Technology",
    "Suggested Vacancies": "Network Security/Application Security",
  },
  {
    "Area of Interest": "(as per the ebove)",
    "Field of Study": "Others ",
    "Suggested Vacancies": "(to be as er the area of interest and suggested vacancy)",
  },
]
`;
